files/external_node_v2.rb

```diff
@@ -7,9 +7,14 @@
 # If --push-facts is given as the only arg, it uploads facts for all hosts and then exits.
 # Useful in scenarios where the ENC isn't used.
 
+require 'rbconfig'
 require 'yaml'
 
-$settings_file = "/etc/puppet/foreman.yaml"
+if RbConfig::CONFIG['host_os'] =~ /freebsd|dragonfly/i
+  $settings_file = "/usr/local/etc/puppet/foreman.yaml"
```

Comment:
Think I got it:
```
/usr/local/bin/ruby21 -I/tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib:/tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-support-3.1.2/lib /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/exe/rspec --pattern spec/\{classes,defines,unit,functions,hosts,integration\}/\*\*/\*_spec.rb --color
/tmp/puppet-foreman/spec/unit/foreman_external_node_spec.rb:14:in `eval': unexpected invocation: Psych.load_file('/usr/local/etc/puppet/foreman.yaml') (Mocha::ExpectationError)
satisfied expectations:
- allowed any number of times, not yet invoked: Psych.load_file('/dev/null')
- allowed any number of times, not yet invoked: Psych.load_file('/etc/puppet/foreman.yaml')
	from /tmp/puppet-foreman/spec/unit/foreman_external_node_spec.rb:14:in `eval'
	from /tmp/puppet-foreman/spec/unit/foreman_external_node_spec.rb:14:in `<class:Enc>'
	from /tmp/puppet-foreman/spec/unit/foreman_external_node_spec.rb:4:in `<top (required)>'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib/rspec/core/configuration.rb:1105:in `load'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib/rspec/core/configuration.rb:1105:in `block in load_spec_files'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib/rspec/core/configuration.rb:1105:in `each'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib/rspec/core/configuration.rb:1105:in `load_spec_files'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:96:in `setup'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:84:in `run'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:69:in `run'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:37:in `invoke'
	from /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/exe/rspec:4:in `<main>'
/usr/local/bin/ruby21 -I/tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/lib:/tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-support-3.1.2/lib /tmp/puppet-foreman/vendor/bundle/ruby/2.1/gems/rspec-core-3.1.7/exe/rspec --pattern spec/\{classes,defines,unit,functions,hosts,integration\}/\*\*/\*_spec.rb --color failed
```
However, as the tests in travis will run on Linux anyway that's not really relevant?